lightning-persister/src/lib.rs

```diff
@@ -187,18 +164,39 @@ impl<ChannelSigner: Sign> chainmonitor::Persist<ChannelSigner> for FilesystemPer
 	// even broadcasting!
 
 	fn persist_new_channel(&self, funding_txo: OutPoint, monitor: &ChannelMonitor<ChannelSigner>, _update_id: chainmonitor::MonitorUpdateId) -> Result<(), chain::ChannelMonitorUpdateErr> {
-		let filename = format!("{}_{}", funding_txo.txid.to_hex(), funding_txo.index);
-		util::write_to_file(self.path_to_monitor_data(), filename, monitor)
+		let key = format!("monitors{}{}_{}", MAIN_SEPARATOR, funding_txo.txid.to_hex(), funding_txo.index);
```

Comment:
I was thinking lets not boterh directly implementing `Persist` or `Persister` at all for the `FilesystemPersister`, instead opting to implement them automatically for *any* `KVStorePersister`.

Reply:
ah that's a good idea. only downside I can think of is that it removes control of the keys used from an implementor.  I guess they could always override the implementation? I'm not sure how/if that would work though